grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -83,36 +86,62 @@ class StockMovementController {
 
-            if (stockMovement.isFromOrder) {
-                redirect(action: "createCombinedShipments", params: params)
-            } else if (stockMovement.requisition?.sourceType == RequisitionSourceType.ELECTRONIC) {
-                if (stockMovement.requisition?.status == RequisitionStatus.CREATED) {
-                    redirect(action: "createRequest", params: params)
-                } else {
+        StockMovementDirection stockMovementDirection = stockMovement?.getStockMovementDirection(currentLocation)
+
+        // Redirect logic for the e-request
+        if (stockMovement?.electronicType) {
+            switch (stockMovementDirection) {
+                case StockMovementDirection.OUTBOUND:
                     redirect(action: "verifyRequest", params: params)
-                }
-            } else {
-                redirect(action: "createInbound", params: params)
+                    break
+                case StockMovementDirection.INBOUND:
+                    User currentUser = AuthService.currentUser
+                    RequisitionStatus requisitionStatus = stockMovement.requisition?.status
+                    // If a request's status is created or a user is a requestor logged in, in a location
+                    // with submit request and without manage inventory, and the request has waiting for approval
+                    // status, we want to redirect to add items page (OBPIH-5981)
+                    if (requisitionStatus == RequisitionStatus.CREATED
+                        || (userService.isUserRequestor(currentUser)
+                            && currentLocation.downstreamConsumer
+                            && requisitionStatus == RequisitionStatus.PENDING_APPROVAL)) {
+                                redirect(action: "createRequest", params: params)
+                                return
+                    }
+                    redirect(action: "verifyRequest", params: params)
+                    break
+                default:
+                    break
             }
+            return
         }
-        else {
-            if (stockMovement.isFromOrder) {
-                redirect(action: "createCombinedShipments", params: params)
-            }
-            redirect(action: "createOutbound", params: params)
+
+        // Redirect logic for the combined shipments feature
+        if (stockMovement.isFromOrder) {
+            redirect(action: "createCombinedShipments", params: params)
+            return
+        }
+
+        // Redirect logic for the regular stock movement feature
+        switch (stockMovementDirection) {
+            case StockMovementDirection.OUTBOUND:
+                redirect(action: "createOutbound", params: params)
+                break
+            case StockMovementDirection.INBOUND:
+                redirect(action: "createInbound", params: params)
+                break
+            default:
+                break
```

Comment:
Same here, I think a default redirect should be present here.
`getStockMovementDirection` can return a `null` value and there is no redirect handler at the end, so I think the app would crash because no redirects are specified at the end.